checker.cpp

```diff
@@ -1,22 +1,41 @@
 #include <assert.h>
 #include <iostream>
+#include "checker.hpp"
 using namespace std;
-
-bool batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
+using namespace BatteryStatus;
+bool Battery::batteryIsOk(std::vector<float>BatteryFunctionParameters) {
+	int i = 0;
+	for (const auto& funptr : functptr) {
+        if(!funptr(BatteryFunctionParameters[i++]))
+		{
+		 return false;
+		}
+    }
+	return true;
+}
+bool Battery::batteryTemperatureIsOk(float temperature)
+{
+	if(temperature < 0 || temperature > 45) {
```

Comment:
Similar code is being used in SOC too. There is scope for reducing duplicate code !